tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -177,6 +183,9 @@ def montecarlo_main_loop(
     virt_packet_last_line_interaction_in_id = []
     virt_packet_last_line_interaction_out_id = []
 
+    # Configuring the Tracking for R_Packets 
+    tracked_rpacket = RPacketCollection()
```

Comment:
You instantiate this outside the main loop.  Will this still work with multi-threading enabled?

Reply:
I haven't tested the multi thread performance for this PR, thus I am not sure about the multi-threading capabilities of this feature. Need to test this & make appropriate changes 🤔

Reply:
No this is not working ... Tests are failing if we try to instantiate this outside the `montecarlo_main_loop`

Reply:
Yes, with the current implementation for the instantiation of the `RPacketCollection()` outside the main loop, the multi threading setting allows it to work. It is not working other way around 🤔